src/storage/secondary/block.rs

```diff
@@ -53,7 +57,10 @@ pub trait BlockBuilder<A: Array> {
     /// Get estimated size of block. Will be useful on runlength or compression encoding.
     fn estimated_size(&self) -> usize;
 
-    /// Get distinct values count of block
+    /// Get the size to append one data into the block
+    fn size_to_append(&self, item: &Option<&A::Item>) -> usize;
```

Comment:
It seems that this function is solely used for determining `should_finish`. However, `should_finish` doesn't need to be accurate -- we can finish early, or slightly exceed the block size. Therefore, I suggest removing this function.

Reply:
So for `should_finish` of `RleBlockBuilder`, will it be ok that only adding `std::mem::size_of::<u16>()` when really append an item for `self.block_builder`?

Reply:
I would prefer first check if the child block builder “should finish”, if not, simply add u16 (for a single RLE count).

Reply:
Currently, for `RleBlockBuilder`, we don't care about the target size of child block builder because it's hard to predict when using RLE, and we only care about the total size together with `rle_counts`. If we first check whether the child block builder "should finish", then we need to set a target size for `RleBlockBuilder` and child block builder respectively.

Reply:
After a second though, I think we should go simple here, just return `self.block_builder.should_finish(item)`.

It is true that this is not accurate -- when there are no items of the same in block, we will produce larger block; when there are a lot of same large items, it will produce smaller block. But this function will return true eventually.

Therefore, we don't need to care about the exact size.

What do you think?

Reply:
My point is that, the `append_one_by_one` function will:

1. first check `should_finish`.
2. then append.

If we have some complex operations, like doing comparison in `should_finish`, the comparison will be done twice in (1) and (2). So I just want the `should_finish` function to be as simple as possible...